php-checks/src/main/java/org/sonar/php/checks/CountInsteadOfEmptyCheck.java

```diff
@@ -89,6 +92,99 @@ private boolean isCountFunction(FunctionCallTree tree) {
     return FUNCTION_PREDICATE.test(TreeValues.of(tree, context().symbolTable()));
   }
 
+  private static final Tree.Kind[] USED_STATEMENTS = {
+    Tree.Kind.EXPRESSION_STATEMENT,
+    Tree.Kind.RETURN_STATEMENT,
+    Tree.Kind.IF_STATEMENT,
+    Tree.Kind.WHILE_STATEMENT,
+    Tree.Kind.DO_WHILE_STATEMENT,
+    Tree.Kind.FOR_STATEMENT,
+    Tree.Kind.FOREACH_STATEMENT,
+    Tree.Kind.SWITCH_STATEMENT,
+    Tree.Kind.ASSIGNMENT,
+    Tree.Kind.FUNCTION_DECLARATION,
+    Tree.Kind.METHOD_DECLARATION};
+
+  private static final Tree.Kind[] AND_OR_TREE = {Tree.Kind.CONDITIONAL_AND, Tree.Kind.CONDITIONAL_OR, Tree.Kind.ALTERNATIVE_CONDITIONAL_AND, Tree.Kind.ALTERNATIVE_CONDITIONAL_OR};
+
+  // Checks if empty() is used on the same variable in the surrounding expression context
+  // Returns true if empty() call is found, indicating we should skip raising an issue
+  private boolean isEmptyUsedInExpression(FunctionCallTree countCallTree, @Nullable ExpressionTree countArgument) {
+    // Early return if count() has no argument
+    if (countArgument == null) {
+      return false;
+    }
```

Comment:
Comments are often very helpful and I like your attempt to explain your code 😊, however, they should be complementary. I think this one just repeats what the code tells us and, hence, is superfluous.
If you want to keep it, I would suggest to refine it as follows:
```suggestion
    if (countArgument == null) {
      // "count" is called without an argument
      return false;
    }
```

Reply:
I think you meant another if condition because the one you are referring to here is removed. I added the comment to the updated condition.